cmd/config_helper.go

```diff
@@ -64,6 +74,22 @@ func toConfig(install, forceIgnored []*api.Package, targets []string, cmdline []
 		sortedPackages = append(sortedPackages, pkg)
 	}
 
+	// now for non requested packages make sure we don't get cycles
+	for _, name := range sortedKeys(allPackages) {
+		pkg := allPackages[name]
+		if _, requested := requested[name]; requested {
+			continue
+		}
+
+		pkg.SetDependencies(nil)
+
+		sortedPackages = append(sortedPackages, pkg)
+	}
+
+	slices.SortFunc(sortedPackages, func(a, b *bazeldnf.RPM) int {
+		return cmp.Compare(a.Name, b.Name)
```

Comment:
isn't the sorting already implicit by virtue of having constructed sortedPackages with the sorted keys?